pkg/dex/server.go

```diff
@@ -0,0 +1,157 @@
+package dex
+
+import (
+	"encoding/json"
+	"net"
+	"net/http"
+	"os"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/gorilla/mux"
+)
+
+type Server struct {
+	sockPath string
+	driver   Driver
```

Comment:
wait why does the dex server have a driver? i thought it would be able to talk to many other drivers, rather than have one itself

does this mean each possible dex implementation has its own server?

Reply:
is the idea that a driver implements the dex api and then we serve it like this?